core/src/test/java/org/bitcoinj/core/LegacyAddressTest.java

```diff
@@ -243,14 +241,22 @@ public void comparisonGreaterThan() throws Exception {
     }
 
     @Test
-    public void comparisonBytesVsString() throws Exception {
-        // TODO: To properly test this we need a much larger data set
-        LegacyAddress a = LegacyAddress.fromBase58(MAINNET, "1Dorian4RoXcnBv9hnQ4Y2C1an6NJ4UrjX");
-        LegacyAddress b = LegacyAddress.fromBase58(MAINNET, "1EXoDusjGwvnjZUyKkxZ4UHEf77z6A5S4P");
-
-        int resultBytes = a.compareTo(b);
-        int resultsString = a.toString().compareTo(b.toString());
-        assertTrue( resultBytes < 0 );
-        assertTrue( resultsString < 0 );
+    public void comparisonBytesVsString() throws IOException {
+        List<String> addrs = new LinkedList<>();
+        List<String> addrs1;
+        List<String> addrs2;
+        BufferedReader dataSetReader = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream("LegacyAddressTestDataset.txt")));
+        String line;
+        while ((line = dataSetReader.readLine()) != null) addrs.add(line);
```

Comment:
I think it would be easier to loop only once and do all the tests while reading the file.